test/remove.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var jsc = require('jsverify');
+
+
+test('remove', function() {
+
+  eq(typeof S.remove, 'function');
+  eq(S.remove.length, 2);
+  eq(S.remove.toString(), 'remove :: String -> StrMap a -> StrMap a');
+
+  function testBasic(key, strMap, strMapResult) {
+    eq(S.remove(key, strMap), strMapResult);
+  }
+
+  function testDoubleRemove() {
+    jsc.assert(jsc.forall(jsc.string, jsc.dict(jsc.number), function(key, map) {
+      var remove = S.remove(key);
+      return S.equals(remove(remove(map)), remove(map));
+    }), {tests: 1000});
+  }
+
+  function testRemove(key, strMap, strMapResult) {
+    testBasic(key, strMap, strMapResult);
+    testDoubleRemove(key, strMap);
+  }
+
+  testRemove('a', {}, {});
+  testRemove('a', {a: 1}, {});
+  testRemove('a', {b: 2}, {b: 2});
+  testRemove('b', {a: 1, b: 2}, {a: 1});
+  testRemove('a', {b: 2, c: 3}, {b: 2, c: 3});
+  testRemove('b', {a: 1, b: 2, c: 3}, {a: 1, c: 3});
+
+  // test inherited properties
+  function Point() {}
+  Point.prototype.x = 0;
+  Point.prototype.y = 0;
+
+  eq(S.remove('x', new Point()), {y: 0});
+  eq(S.remove('y', new Point()), {x: 0});
+  eq(S.remove('z', new Point()), {x: 0, y: 0});
```

Comment:
Shall we include similar test cases for `insert`?

Reply:
Makes sense